internal/manifest/oci_index.go

```diff
@@ -84,6 +84,45 @@ func (index *OCI1IndexPublic) UpdateInstances(updates []ListUpdate) error {
 	return nil
 }
 
+func (index *OCI1Index) EditInstances(updates []ListEdit) error {
+	if len(updates) != len(index.Manifests) {
+		//return fmt.Errorf("incorrect number of update entries passed to OCI1Index.UpdateInstances: expected %d, got %d", len(index.Manifests), len(updates))
+		if len(updates) > len(index.Manifests) {
```

Comment:
https://github.com/containers/image/pull/1875/files#r1128482848

Reply:
Added a function below to remove element, it should resize the slice as well.

Reply:
We do need the delete operation, eventually. I’m not sure that we need it _now_ (though adding it, by itself, wouldn’t hurt), and more importantly that doesn’t resolve the issues with this code branch.

- The API documentation still says the number of updates must match (which is what the _original_ review comment was pointing out.)
- Adding `ListOpRemove` directly causes the `newIndexes` reallocation to be incorrect, because `len(updates)` now counts `ListOpRemove` items.
- The update implementation is now _also_ broken by this: e.g. a two-item manifest list with entries (`A`, `B`), and edits (`ListOpRemove`, `ListOpUpdate(B')`), which _should_ ideally result in a single-item (`B'`), instead results in (`B`, `B'`), because the indices get out of sync (and if the `newIndexes` allocation size were computed correctly, it would cause a panic due to out-of-bounds indexing).
- Also consider how much work it would be to remove one instance but keep all others unchanged: the caller would have to create a `[]ListEdit` with N-1 `ListOpIgnore` entries and one `ListOpRemove` entry, when ideally it, IMHO, should be just a single “remove that one instance“ instruction.

I think we definitely need a design that can be extended to support deletions; but I don’t really think that’s _this_ design, so dealing with a correct _implementation_ of deletions seems to be a bit premature. Or, maybe, do you want the code in precisely so that we can game out all the design and implementation concerns, even at the cost of extra coding effort?

Reply:
>  I’m not sure that we need it now (though adding it, by itself, wouldn’t hurt)

I think we don't need `delete` as of now but I thought why not add it early, OTOH its just increasing the coding overhead should we do `delete` later on one a minimal MVP of the API is already in and working ?

Reply:
I really don’t mind either way, entirely up to you.

If deletion is a part of this PR, it will need to work, correctly, and probably with tests. That’s extra effort, but having the code makes it easier to discuss / imagine the parts of the API that we do need right now, sure, why not. Personally I think I can imagine these parts of the design well enough without actually preparing a production-quality quality.

Reply:
Okay I have removed `delete` for now to speed things up, I think it'll be easier to add it later on once major infrastructure is built.